kratos/mpi/tests/test_utilities/mpi_cpp_test_utilities.cpp

```diff
@@ -172,7 +174,101 @@ void MPICppTestUtilities::GenerateDistributedBarStructure(
         }
     }
 
-    // Compute communicaton plan and fill communicator meshes correctly
+    // Compute communication plan and fill communicator meshes correctly
+    auto filler = ParallelFillCommunicator(rModelPart, rDataCommunicator);
+    filler.Execute();
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void MPICppTestUtilities::GenerateDistributedTriangleMesh(
+    ModelPart& rModelPart,
+    const DataCommunicator& rDataCommunicator
+    )
+{
+    // Set MPI coomunicator
+    ModelPartCommunicatorUtilities::SetMPICommunicator(rModelPart, rDataCommunicator);
+
+    // Add variables
+    rModelPart.AddNodalSolutionStepVariable(PARTITION_INDEX);
+
+    // Create properties
+    auto p_prop = rModelPart.CreateNewProperties(1, 0);
+
+    // MPI data
+    const int rank = rDataCommunicator.Rank();
+    const int world_size = rDataCommunicator.Size();
```

Comment:
does this work for any number of cores? I dont see any checks

Reply:
The checks are later